proto-lens-tests/tests/repeated_test.hs

```diff
@@ -5,24 +5,48 @@
 -- https://developers.google.com/open-source/licenses/bsd
 
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 module Main where
 
-import Proto.Repeated
-import Proto.Repeated_Fields
-import Test.Framework (testGroup)
-import Data.ProtoLens
-import Lens.Family2 ((&), (.~))
 import Data.ByteString.Builder (byteString)
 import Data.Monoid ((<>))
+import Data.Proxy (Proxy(..))
+import Lens.Family2 (Lens', (&), (.~), view, set)
+import Test.Framework (testGroup)
+import Test.QuickCheck
+import qualified Data.Text as T
+import qualified Data.Vector.Generic as V
 
+import Data.ProtoLens
+import Data.ProtoLens.Arbitrary
 import Data.ProtoLens.TestUtil
+import Proto.Repeated
+import Proto.Repeated_Fields
 
 defFoo :: Foo
 defFoo = defMessage
 
 defBar :: Bar
 defBar = defMessage
 
+-- | Test that the vector and list lenses for repeated fields
+-- behave the same.
+vectorTest ::
+    forall a v b . (Eq a, Eq b, Show a, Show b, Message a, V.Vector v b)
+    => String -> Proxy a -> Gen b
+    -> Lens' a [b] -> Lens' a (v b) -> Test
+vectorTest name _ arbitraryElem listLens vecLens = testGroup name
```

Comment:
Do you need a `Proxy a` here or can you get by with \@ a ?

Reply:
Switched to `@`.  Nice!